packages/gafl-webapp-service/src/processors/find-permit.js

```diff
@@ -1,45 +1,43 @@
 import filterPermits from './filter-permits.js'
 import crypto from 'crypto'
 import db from 'debug'
-const debug = db('webapp:find-permit')
 
-export const findPermit = async (permission, request) => {
-  /*
-   * To stop repeated reads of the API with users repeatably refreshing the page, the transaction cache stores
-   * a hash of itself. If the transaction cache has not changed the permit is not recalculated.
-   *
-   * The section of the transaction cache subject to the hashing algorithm excludes
-   * name, address, or anything not effecting permit filter
-   */
-  const hashOperand = (({ hash: _hash, permit: _permit, licensee: _licensee, ...p }) => p)(permission)
+const debug = db('webapp:find-permit')
 
-  // To calculate a permit, hash and save
-  const addHashAndPermit = async () => {
-    const permit = await filterPermits(permission)
-    permission.permit = permit
-    if (permit) {
-      if (!permit.newCostStartDate || !permit.newCost) {
-        debug('permit missing new cost details', permission)
-      }
-    } else {
-      debug("permit wasn't retrieved", permission)
+export const retrievePermit = async permission => {
+  const permit = await filterPermits(permission)
+  if (permit) {
+    if (!permit.newCostStartDate || !permit.newCost) {
+      debug('permit missing new cost details', permission)
     }
-
-    const hash = crypto.createHash('sha256')
-    hash.update(JSON.stringify(hashOperand))
-    permission.hash = hash.digest('hex')
-    await request.cache().helpers.transaction.setCurrentPermission(permission)
+  } else {
+    debug("permit wasn't retrieved", permission)
   }
+  return permit
+}
+
+const updateHash = permission => {
+  const { hash: _hash, permit: _permit, licensee: _licensee, ...hashOperand } = permission
+  const hash = crypto.createHash('sha256')
+  hash.update(JSON.stringify(hashOperand))
+  const updatedHash = hash.digest('hex')
+  return updatedHash
+}
 
-  if (!permission.hash) {
-    await addHashAndPermit()
+export const findPermit = async permission => {
+  const permitPermission = { ...permission }
+  if (!permitPermission.hash) {
+    permitPermission.permit = await retrievePermit(permitPermission)
+    permitPermission.hash = updateHash(permitPermission)
   } else {
-    const hash = crypto.createHash('sha256')
-    hash.update(JSON.stringify(hashOperand))
-    if (hash.digest('hex') !== permission.hash) {
-      await addHashAndPermit()
+    const currentHash = updateHash(permitPermission)
+    if (currentHash !== permitPermission.hash) {
+      permitPermission.permit = await retrievePermit(permitPermission)
+      permitPermission.hash = currentHash
     } else {
       debug("permit data present and doesn't need updating")
     }
   }
+
+  return permitPermission
```

Comment:
This is closer, but still not what I asked for. If you think about the name 'findPermit' and don't look at the code, what would you expect it to do? I know that I'd expect it to find and return a permit. I wouldn't expect it to update my passed-in permission with the permit details. I _certainly_ wouldn't expect it to update the hash of the passed-in permission.

So you need to modify this function in light of this (I think that retrievePermit is probably doing what findPermit should be doing) and then modify the calling code to take account of the revised behaviour of findPermit.